polaris/ocean/ice_shelf/ssh_adjustment.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+import xarray as xr
+from mpas_tools.cime.constants import constants
+from mpas_tools.io import write_netcdf
+
+from polaris import Step
+from polaris.ocean.vertical import update_layer_thickness
+
+
+class SshAdjustment(Step):
+    """
+    A step for iteratively adjusting the pressure from the weight of the ice
+    shelf to match the sea-surface height as part of ice-shelf 2D test cases
+    """
+    def __init__(self, component, init, forward, indir=None,
+                 name='ssh_adjust'):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        component : polaris.ocean.Ocean
+            The ocean component that this task belongs to
+
+        init : polaris.Step
+            the step that produced the initial condition
+
+        forward: polaris.Step
+            the step that produced the state which will be adjusted
+
+        indir : str, optional
+            the directory the step is in, to which ``name`` will be appended
+
+        name : str, optional
+            the name of this step
+        """
+        super().__init__(component=component, name=name,
+                         indir=f'{indir}/ssh_adjustment')
+
+        self.add_input_file(filename='final.nc',
+                            work_dir_target=f'{forward.path}/output.nc')
+        self.add_input_file(filename='init.nc',
+                            work_dir_target=f'{forward.path}/init.nc')
+        self.add_input_file(filename='mesh.nc',
+                            work_dir_target=f'{init.path}/culled_mesh.nc')
```

Comment:
I don't think this will work in the generalized case.  For example, this file would not be in the `init` step for ISOMIP+.  So I would think this would be another filename to pass instead of passing `init` itself.